contracts/holder/src/tests.rs

```diff
@@ -0,0 +1,122 @@
+
+use cosmwasm_std::{from_binary, Addr, coins, BankMsg, CosmosMsg, Timestamp};
+use cosmwasm_std::testing::{mock_dependencies, mock_env, mock_info};
+
+use crate::msg::{InstantiateMsg, QueryMsg, ExecuteMsg};
+use crate::contract::{instantiate, query, execute};
+use crate::error::ContractError;
+
+
+// Queried withdrawer address is same as one we instantiated with
+#[test]
+fn test_instantiate_query_withdrawer() {
+    let mut deps = mock_dependencies();
+    let env = mock_env();
+    let test_addr = "anaddressonneutron";
+    let instantiate_msg = InstantiateMsg {
+        withdrawer: Some(test_addr.to_string())
+    };
+
+    instantiate(
+        deps.as_mut(),
+        env.clone(),
+        mock_info("sender", &[]),
+        instantiate_msg
+    )
+    .unwrap();
+
+    let resp = query(deps.as_ref(), env, QueryMsg::Withdrawer {}).unwrap();
+    let resp: Addr = from_binary(&resp).unwrap();
+
+    // assert that instantiated address = queried address
+    assert_eq!(resp, Addr::unchecked("anaddressonneutron"));
+
+    // and for posterity check that a different address fails
+    assert_ne!(resp, Addr::unchecked("anotheraddress"));
+}
+// Test withdraw
+// 1. A sender who is not the withdrawer is unauthorized
+// 2. Authorized withdrawer can withdraw partial amount
+// 3. Authorized withdrawer can withdraw complete balance
+
+#[test]
+fn test_withdraw() {
```

Comment:
I think it would be nice to split this test up into smaller ones to test out each error possible cases

Reply:
I rewrote the entire test suite with a suite builder and more atomic tests. Thanks for the suggestion, much prefer the way it is after the change.